packages/forms/src/patterns/date-of-birth/date-of-birth.ts

```diff
@@ -0,0 +1,110 @@
+import * as z from 'zod';
+
+import { type DateOfBirthProps } from '../../components.js';
+import {
+  type Pattern,
+  type PatternConfig,
+  validatePattern,
+} from '../../pattern.js';
+import { getFormSessionValue } from '../../session.js';
+import {
+  safeZodParseFormErrors,
+  safeZodParseToFormError,
+} from '../../util/zod.js';
+
+const configSchema = z.object({
+  label: z.string().min(1),
+  required: z.boolean(),
+  hint: z.string().optional(),
+});
+
+export type DateOfBirthPattern = Pattern<z.infer<typeof configSchema>>;
+
+export type DateOfBirthPatternOutput = z.infer<
+  ReturnType<typeof createDOBSchema>
+>;
+
+export const createDOBSchema = (data: DateOfBirthPattern['data']) => {
+  const daySchema = z
+    .string()
+    .regex(/^\d{1,2}$/, 'Invalid day format')
+    .optional();
+  const monthSchema = z
+    .string()
+    .regex(/^(0[1-9]|1[0-2])$/, 'Invalid month format')
+    .optional();
+  const yearSchema = z
+    .string()
+    .regex(/^\d{4}$/, 'Invalid year format')
+    .optional();
```

Comment:
These schemas look correct, but I think we may want them to be coerced into numbers. @ethangardner thoughts?

Reply:
I am going to merge the current PR to unblock my next work, but in some point if want to update anything here I can raise a separate PR for it.